modules/nw-cfg-tuning-interface-cni.adoc

```diff
@@ -0,0 +1,117 @@
+// Module included in the following assemblies:
+//
+// * networking/hardware_networks/configuring-sriov-operator.adoc
+:_content-type: PROCEDURE
+[id="nw-configuring-tuning-cni_{context}"]
+= Configuring the tuning CNI
+
+The following procedure shows an example of configuring the tuning CNI to change the interface level network `net.ipv4.conf.IFNAME.accept_redirects` sysctl. This example enables the acceptance and sending of ICMP redirected packets.
+
+.Procedure
+
+. Create a network attachment definition, such as `tuning-example.yaml`, with the following content:
++
+[source,yaml]
+----
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: tuningnad <1>
+  namespace: default <2>
+spec:
+  config: '{
+  "cniVersion": "0.4.0", <3>
+   "name": "tuningnad", <4>
+   "plugins": [
+     {
+       "type": "bridge", <5>
+     },
+     {
+       "type": "tuning", <6>
+       "sysctl": {
+            "net.ipv4.conf.IFNAME.accept_redirects": "1" <7>
+        }
+      }
+     ]
+}'
+----
+<1> The name for the additional network.
+<2> The namespace that the object is associated with.
+<3> The CNI specification version.
+<4> The value for the name parameter you provided previously for the CNI configuration.
+<5> The name of the main CNI plug-in to configure. The `bridge` network plug-in is an example.  
+<6> The value of the CNI meta plug-in.
+<7> The sysctl to set.
+
+. Apply the yaml:
++
+[source,terminal]
+----
+$ oc apply -f tuning-example.yaml
+----
++
+.Example output
+[source,terminal]
+----
+networkattachmentdefinition.k8.cni.cncf.io/tuningnad created
+----
+
+. Create a pod such as `examplepod.yaml` with the network attachment definition.
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: tunepod
+  namespace: default
+  annotations:
+    k8s.v1.cni.cncf.io/networks: tuningnad <1>
+spec:
+  containers:
+  - name: podexample
+    image: centos
+    command: ["/bin/bash", "-c", "sleep INF"]
+----
+<1> Specify the name of the configured NetworkAttachmentDefinition.
+
+. Apply the yaml:
++
+[source,terminal]
+----
+$ oc apply -f examplepod.yaml
+----
+
+. Verify that the pod is created:
++
+[source,terminal]
+----
+$ oc get pod
+----
++
+.Example output
+[source,terminal]
+----
+NAME      READY   STATUS    RESTARTS   AGE
+tunepod   1/1     Running   0          47s
+----
+
+. Log in to the pod:
++
+[source,terminal]
+----
+$ oc rsh tunepod
+----
+
+. Verify the values of the configured sysctl flags. For example find the value `net.ipv4.conf.net1.accept_redirects` as follows:
++
+[source,terminal]
+----
+sh-4.4# sysctl -a | grep net.ipv4.conf.net1.accept_redirects
```

Comment:
Same here. If we want to avoid "permission denied"  messages:

```
ysctl -a | grep net.ipv4.conf.net1.accept_redirects
sysctl: permission denied on key 'fs.protected_fifos'
sysctl: permission denied on key 'fs.protected_hardlinks'
sysctl: permission denied on key 'fs.protected_regular'
sysctl: permission denied on key 'fs.protected_symlinks'
sysctl: permission denied on key 'kernel.cad_pid'
sysctl: permission denied on key 'kernel.usermodehelper.bset'
sysctl: permission denied on key 'kernel.usermodehelper.inheritable'
net.ipv4.conf.net1.accept_redirects = 1
sysctl: permission denied on key 'net.ipv4.tcp_fastopen_key'
sysctl: permission denied on key 'net.ipv6.conf.all.stable_secret'
sysctl: permission denied on key 'net.ipv6.conf.default.stable_secret'
sysctl: permission denied on key 'net.ipv6.conf.eth0.stable_secret'
sysctl: permission denied on key 'net.ipv6.conf.lo.stable_secret'
sysctl: permission denied on key 'net.ipv6.conf.net1.stable_secret'
sysctl: permission denied on key 'vm.mmap_rnd_bits'
sysctl: permission denied on key 'vm.mmap_rnd_compat_bits'
sysctl: permission denied on key 'vm.stat_refresh'
```

Better to use:

```
sh-4.4$ sysctl net.ipv4.conf.net1.accept_redirects          
net.ipv4.conf.net1.accept_redirects = 1
```